python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -529,6 +540,74 @@ def resolve_type(self, _graphene_info) -> Optional[str]:
                     )
         return None
 
+    def resolve_freshnessInfo(self, graphene_info) -> Optional[GrapheneAssetFreshnessInfo]:
+        from dagster import DagsterEventType, EventRecordsFilter
+        from dagster._utils.calculate_data_time import get_upstream_materialization_times_for_record
+
+        # no FreshnessPolicy, so freshnessInfo does not apply
```

Comment:
maybe we can break this out into a helper method in a separate file that returns `GrapheneAssetFreshnessInfo` objects to keep the `GrapheneAssetNode` object short?